aioboto3/s3/inject.py

```diff
@@ -263,7 +263,7 @@ async def file_reader() -> None:
         nonlocal exception
         part = 0
         eof = False
-        while not eof:
+        while not exception and not eof:
```

Comment:
So if an exception is caught in `file_reader`, `eof` will be set to true, so this achieves nothing.

On an exception `await file_reader_future` should finish, the asyncio.wait, should return `exception_event.wait()` as the first completeled coroutine as the `uploader()` coroutines should instantly exit as soon as exception is set, call abort multipart, cancel any running coroutines and then raise an exception.

Reply:
I hear what you are saying, but this indeed resolved the bug on my side. I can double check one more time.

Reply:
For clarification, when uploader encounters an exception, then we want to abort file_reader. That's why while not exception is needed.

Reply:
Ahh yes, 💯, yeah if an uploader encounters an error the file_reader would continue. Makes total sense as if any of the uploading coroutines get upset, it wont interrupt the reading of the file.

Reply:
Great! Thank you!